File: src/3105.longest-strictly-increasing-or-strictly-decreasing-subarray.3372/solution.ts
function longestMonotonicSubarray (nums: number[]): number {
    let result = 1;
    let increase = 1;
    let decrease = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > nums[i - 1]) {
            increase++;
            decrease = 1;
        } else if (nums[i] < nums[i - 1]) {
            decrease++;
            increase = 1;
        } else {
            increase = 1;
            decrease = 1;
        }
        result = Math.max(result, increase, decrease);
    }

    return result;
}

File: src/3105.longest-strictly-increasing-or-strictly-decreasing-subarray.3372/solution.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int result = 1;
        int increase = 1;
        int decrease = 1;
        for(int i=1;i< nums.length; i++){
            if(nums[i]>nums[i-1]){
                increase++;
                result = Math.max(result,increase);
            }else{
                increase = 1;
            }
            if(nums[i]<nums[i-1]){
                decrease++;
                result = Math.max(result,decrease);
            }else{
                decrease = 1;
            }
            
        }
        return result;
    }
}